src/Autonomous.cpp

```diff
@@ -412,30 +416,39 @@ void Autonomous::autonomyIter()
       map.addPoints(toTransform(pose), lidar_scan, mapDoesOverlap ? 0.4 : 0);
       mapDoesOverlap = lidar_scan.size() > mapOverlapSampleThreshold;
     }
-    if (plan_cost >= INFINITE_COST || ++time_since_plan % REPLAN_PERIOD == 0) {
-      // TODO planning should happen in a separate thread
-      // because it takes a long time
-      point_t point_t_goal;
-      point_t_goal.topRows(2) = plan_target.topRows(2);
-      point_t_goal(2) = 1.0;
-      point_t_goal = toTransform(pose) * point_t_goal;
-      //plan_t new_plan = getPlan(lidar_scan, point_t_goal, PLANNING_GOAL_REGION_RADIUS);
-      plan_t new_plan = getPlan([&](double x, double y, double radius)->bool {
-        // transform the point to check into map space
-        point_t relPoint = {x, y, 1};
-        point_t p = invTransform * relPoint;
-        return map.hasPointWithin(p, radius);
-      }, point_t_goal, PLANNING_GOAL_REGION_RADIUS);
-      double new_plan_cost = planCostFromIndex(new_plan, 0);
-      // we want a significant improvement to avoid thrash
-      log(LOG_DEBUG, "old cost %f, new cost %f\n", plan_cost, new_plan_cost);
-      if (new_plan_cost < plan_cost * 0.8) {
-        plan_idx = 0;
-        plan_base = pose;
-        plan_cost = new_plan_cost;
-        plan = new_plan;
-        time_since_plan = 0;
-      }
+    if ((plan_cost >= INFINITE_COST || ++time_since_plan % REPLAN_PERIOD == 0)) {
```

Comment:
Since planning runs in a separate thread now, I think we should increase the frequency at which we replan by either removing this condition or decreasing `REPLAN_PERIOD`. I ran some tests in the simulator with this condition removed, and the computed plans were significantly better, responding to changes in `plan_target` faster and resulting in more optimal paths.